debugging-contd/problem-set-debugging-contd.md

```diff
@@ -230,72 +191,108 @@ Some examples of things to try:
 
 ##### !end-explanation
-
 ### !end-challenge
-
 <!-- prettier-ignore-end -->
-<!--END CHALLENGE-->
 
-<!--BEGIN CHALLENGE-->
 <!-- prettier-ignore-start -->
-
 ### !challenge
-
 * type: paragraph
 * id: 4b1ba8ea-b9e5-4c37-b486-17fd865bc17b
 * title: Debugging Continued
-
 ##### !question
 
-Moira knows what's happening and why it's happening.  She's not quite sure yet how to fix her code, though!  What are some debugging steps she can take as she works on a solution?
+Dahlia knows what's happening and why it's happening. What code does she need to add or change to ensure the last base and its count gets added to the `encoding` list before we call join in our `return` statement?
 
 ##### !end-question
-
-<!--optional-->
 ##### !explanation
 
-Moira tries several things to debug her code.  
-- Verify Library Functions: Moira starts by reading the count_neighbors function and makes sure she understands what it's doing.  
-- Check for Logical Errors: Moira realizes that she's changing the stage and using it in count_neighbors to count the previous generation's neighbors at the same time.  She realizes she needs two versions of the stage.  
-- Try Something: Moira tries several different things to come up with a solution that allows her to check the previous version of the stage and update the new version at the same time.
+There are other ways Dahlia could solve this issue, but she chose to add 2 lines of code to the function `encode_strand`. After the loop ends, Dahlia has added the following lines before the `return` statement:
 
-Take some time to try to debug Moira's code.  When you are ready, take a look at [Moira's solution here](https://replit.com/@adacore/DebuggingContdProblemSetSolution).
+```py
+new_entry = strand[-1] + str(count)
+encoding.append(new_entry)
+```
 
 ##### !end-explanation
-
 ### !end-challenge
+<!-- prettier-ignore-end -->
 
+## Part 2 Questions
+
+In Part 2, Dahlia has refactored her code to first compress the strings, then categorize them. 
+
+Before you answer these questions, ensure you have followed all instructions in the `PROBLEM_SET_README.md` file for Part 2 to change the imports in `main.py`, as well as uncommented the Part 2 tests in the `sequence_tests.py` file.
+
+<!-- prettier-ignore-start -->
+### !challenge
+* type: paragraph
+* id: f9f85ba0-3390-46c0-9244-38b1bffa6824
+* title: Debugging Continued
+##### !question
+
+When Dahlia runs her code with the `r` option, no errors are raised, but she gets a test failure when running with the `t` option.
+
+Take a few minutes and use some debugging techniques to find out why the Part 2 code is behaving the way it is.
+
+List a few of the things you tried and what you discovered.
+
+##### !end-question
+##### !explanation
+
+Some examples of things to try:
+
+- Add print statements that print out `stage` at the beginning and the end of `one_generation`.
+
+- Add print statements inside of the for loops that print out `stage`.
```

Comment:
Since these reference Game of Life, we could update them to be like:

- Add a print statement to the test just before the assert statement to see the value of `result`.